stats/influxdb/collector.go

```diff
@@ -107,11 +107,31 @@ func (c *Collector) commit() {
 		return
 	}
 
+	cache := map[*stats.SampleTags]struct {
+		tags   map[string]string
+		values map[string]interface{}
+	}{}
+
+	var tags map[string]string
+	var values map[string]interface{}
 	for _, sample := range samples {
+		if cached, ok := cache[sample.Tags]; ok {
+			tags = cached.tags
+			values = cached.values
+			values["value"] = sample.Value // Overwrite the "value" field
+		} else {
+			tags = sample.Tags.CloneTags()
+			values = c.extractFields(tags)
+			values["value"] = sample.Value // Ok since the "value" field will always be overwritten
```

Comment:
See comment above, the whole `values` generation should probably be outside of the `if {} else {}` block